packages/pigeon/e2e_tests/test_objc/ios/Runner/MyApi.m

```diff
@@ -6,7 +6,10 @@
 #import <Flutter/Flutter.h>
 
 @implementation MyApi
-- (ACSearchReply *)search:(ACSearchRequest *)input error:(FlutterError **)error {
+- (void)initialize:(FlutterError *_Nullable *_Nonnull)error {
```

Comment:
Why is an empty `+initialize` being generated now?

Reply:
I didn't add the parameter name for functions that have zero arguments.  This is declared as `void initialize()`.  Since no parameter name is given one isn't inserted in the selector.  `initialize:` versus `initializeError:` or `initializeCompletion:`

Reply:
This question was PEBCAK; I saw the `+` at the start of the line` and the word `initialize`, and somehow ignored everything else and thought this was generating an `[NSObject +initialize]` for no apparent reason.

Now that I understand what I'm looking at though: for zero-length we correctly generate these as `...WithError:` and `...WithCompletion:`. Is there a reason not to do that? I could see "consistency, because we don't do `With` in other cases", except that this is already inconsistent with other cases, so why no be inconsistent and right rather than inconsistent and wrong? (And I would argue that `...WithError:` is actually significantly more consistent with `...:error:` than having no name for error in the zero-argument case.)

Reply:
`..WithError:` is more idiomatic.  I think we could do that with a tiny change to `_getSelectorComponents`.  I'm on board as long  as I don't have to replumb everything.  Do you think it should be like that?  I think we are running into the question again about consistency between platforms and more idiomatic objc code.  Coupled with the fact that consistency has usually meant less complex generation code.

Reply:
I implemented `...WithError:`, it wasn't so bad.  There is some cleanup that's probably possible to make it better but it wasn't apparent to me right now.  PTAL.